allennlp/data/dataset_readers/vqav2.py

```diff
@@ -435,11 +443,12 @@ class Split(NamedTuple):
         # It would be much easier to just process one image at a time, but it's faster to process
         # them in batches. So this code gathers up instances until it has enough to fill up a batch
         # that needs processing, and then processes them all.
-        question_dicts = self.shard_iterable(questions["questions"])
-        image_prefix = "COCO_%s_" %questions['data_subtype']
+        question_dicts = list(self.shard_iterable(questions["questions"]))
```

Comment:
Why does this need to be a list?

Reply:
You iterate over it more than once.  The test failed to return all instances because the iterator was getting consumed (in the dataset reader test, and in the model test, I was seeing one instance instead of 3).  I didn't dig into `shard_iterable` to figure out why it was getting consumed, but consuming the whole iterator and making it a list fixed the problem.  Something funny is going on inside `shard_iterable` when you iterate over it more than once.